nodejs/dist/uws.js

```diff
@@ -1,98 +1,134 @@
+        super();
+        /* we currently do only support port as option */
+        this.nativeServer = new NativeServer(options.port);
 
-    /* we currently do only support port as option */
-    this.nativeServer = new NativeServer(options.port);
-    var self = this;
+        this.nativeServer.onConnection((nativeSocket) => {
+            const socket = new Socket(nativeSocket, this);
+            this.nativeServer.setData(nativeSocket, socket);
+            this.emit('connection', socket);
+        });
 
-    this.close = function () {
-        self.nativeServer.close();
+        this.nativeServer.onDisconnection((nativeSocket) => {
+            const socket = self.nativeServer.getData(nativeSocket);
+            socket.emit('close');
+            /* make sure to clear any set data */
+            self.nativeServer.setData(nativeSocket);
+        });
+
+        this.nativeServer.onMessage((nativeSocket, message, binary) => {
+            const socket = self.nativeServer.getData(nativeSocket);
+            socket.emit('message', binary ? message : message.toString());
+        });
     }
-    this.broadcast = function (message, options) {
+
+    /**
+     * Closes the server.
+     *
+     * @public
+     */
+    close() {
+        this.nativeServer.close();
+    }
+
+    /**
+     * Broadcast a message to all sockets.
+     *
+     * @param {String|Buffer} message The message to broadcast
+     * @param {Object} options Broadcast options
+     * @public
+     */
+    broadcast(message, options) {
         /* only listen to binary option */
-        this.nativeServer.broadcast(message, options.binary);
-    };
-
-    /* original connection callback */
-    this.connectionCallback = function (nativeSocket) {
-        var socket = new Socket(nativeSocket, self);
-        self.nativeServer.setData(nativeSocket, socket);
-        self.emit('connection', socket);
-    };
-
-    /* this function needs to handle the upgrade directly! */
-    this.handleUpgrade = function (request, socket, upgradeHead, callback) {
+        this.nativeServer.broadcast(message, options && options.binary || false);
+    }
+
+    /**
+     * Handles a HTTP Upgrade request.
+     *
+     * @param {http.IncomingMessage} request HTTP request
+     * @param {net.Socket} Socket between server and client
+     * @param {Buffer} upgradeHead The first packet of the upgraded stream
+     * @param {Function} callback Callback function
+     * @public
+     */
+    handleUpgrade(request, socket, upgradeHead, callback) {
+        const nativeServer = this.nativeServer;
+
         /* register a special connection handler */
-        self.nativeServer.onConnection(function (nativeSocket) {
-            var socket = new Socket(nativeSocket, self);
-            self.nativeServer.setData(nativeSocket, socket);
+        nativeServer.onConnection((nativeSocket) => {
+            const sock = new Socket(nativeSocket, this);
+            nativeServer.setData(nativeSocket, sock);
```

Comment:
I can't tell, but there may be a small difference in memory use since we are now building a closure around our function to hold that nativeServer value. It also prevents us from (if we would ever want to) check if `this.nativeServer` is actually set or not (post-close).